pkg/collector/collector.go

```diff
@@ -67,3 +76,26 @@ func (b *BaseCollector) Export() error {
 
 	return nil
 }
+
+// Helper function to collect output of a given kubectl command to a file. Returns kubectl's stderr output if stdout output is empty.
+func (b *BaseCollector) collectKubeResourceToFile(rootPath, fileName string, kubeCmds []string) error {
+	resourceFile := filepath.Join(rootPath, fileName)
+	outputStreams, err := utils.RunCommandOnContainerWithOutputStreams("kubectl", kubeCmds...)
+	if err != nil {
+		return err
+	}
+
+	output := outputStreams.Stdout
+	// If kubectl stdout output is empty, i.e., there is no resource of this type within the cluster, the absence of this resource is logged in the file with the relevant message from stderr (Ex: "No resource found...").
+	if len(output) == 0 {
+		output = outputStreams.Stderr
+	}
+
+	if err = utils.WriteToFile(resourceFile, output); err != nil {
```

Comment:
💡Code  thought: I really like the idea of on-failure return early.  That way the `resourceFile := filepath.Join(rootPath, fileName)` don't need to be used until the very end.

Reply:
@Tatsinnit could you expand on what you mean by that?

Reply:
💡 Code comment was to do with 2 key thing first is the readability, flow of code and second error handling. Also, I was thinking that for consistency why not like this:

* Since `resourceFile` is not used at all till it reaches this code? Also t
    * Also could entirely move to separate func with better name which will identify the exact purpose of this check and add it to the flow?

``` 
resourceFile := filepath.Join(rootPath, fileName)
err = utils.WriteToFile(resourceFile, output);
if err != nil {
      // do something
}
```

further: 

* In case you want to have bunch of error handling `err != nil` scenario you could explore common method for this particular work.

Reply:
@Tatsinnit  You're right `resourceFile` should be moved further down the code. I don't think we should separate the function because `utils.WriteToFile()` requires `output` as a parameter

Reply:
💡 Sure, yeah key idea is code cleanliness and readability. feel free to scope it otherwise we can do it latter, key thing is that dividing `functs` to smaller unit usually will help to see, it might helps in UTs (latter as well).